test/Window/WindowBase.test.cpp

```diff
@@ -219,5 +221,33 @@ TEST_CASE("[Window] sf::WindowBase", runDisplayTests())
 
         // Should compile if user provides both a specific handler and a catch-all
         windowBase.handleEvents([](const sf::Event::Closed&) {}, [](const auto&) {});
+
+        // Should compile if user provides a handler taking an event subtype by value or reference,
+        // but not rvalue reference because it would never be called.
+        windowBase.handleEvents([](sf::Event::Closed) {});
+        windowBase.handleEvents([](const sf::Event::Closed) {});
+        windowBase.handleEvents([](sf::Event::Closed&) {});
+        windowBase.handleEvents([](const sf::Event::Closed&) {});
+
+        // Should compile if user provides a move-only handler
+        windowBase.handleEvents([p = std::make_unique<int>()](const sf::Event::Closed&) {});
```

Comment:
Should we store this move-only lambda on the stack and test for the state of the lambda after `handleEvents` is used? Should we expect that unique pointer to be null after it's passed to `handleEvents`?

Reply:
This test is merely making sure that we are not missing an `std::forward` in `handleEvents` implementation. 
If we put this handler on the stack, then `handleEvents` would take it by reference. The `unique_ptr` would be left intact. This test is not run anyways. It is just checking the code compiles.